### PR TITLE
release: v1.29.0

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,68 @@
...
+* Added developer journey tutorial [#6201](https://github.com/GoogleContainerTools/skaffold/pull/6201)
+
+New Features:
+* Implement port forwarding for Docker deployer [#6303](https://github.com/GoogleContainerTools/skaffold/pull/6303)
+* Implement Logger for Docker Deployer [#6265](https://github.com/GoogleContainerTools/skaffold/pull/6265)
+* [v2] Add renderer manifests as dependencies  [#6277](https://github.com/GoogleContainerTools/skaffold/pull/6277)
+* [v2] Enable the helm in `.manifests.generate` (bypass kpt fn) [#6270](https://github.com/GoogleContainerTools/skaffold/pull/
+6270)
+* [v2] Set v2 default renderer and deployer config. [#6252](https://github.com/GoogleContainerTools/skaffold/pull/6252)
+* [v2] Add a new deployer kptV2. [#6235](https://github.com/GoogleContainerTools/skaffold/pull/6235)
```
Comment (tejal29) [inline CHANGELOG.md:20]: Can we remove this from notes. Currently users can't try this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,68 @@
+# v1.29.0 Release - 07/29/2021
+Note: This release comes with a new config version, `v2beta20`. To upgrade your skaffold.yaml, use `skaffold fix`. If you choose not to upgrade, skaffold will auto-upgrade as best as it can.
+
+Highlights:
+* Skaffold master -> main [#6263](https://github.com/GoogleContainerTools/skaffold/pull/6263)
+* Add state to event handler to track `logrus` output events [#6272](https://github.com/GoogleContainerTools/skaffold/pull/6272)
+* Remove skaffoldLogs endpoint, consolidate into events endpoint [#6296](https://github.com/GoogleContainerTools/skaffold/pull/6296)
```
Comment (tejal29) [inline CHANGELOG.md:7]: Event API v2 has not being communicated to users. 
Can we move this to updates section

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,68 @@
...
+* fix: kubeContext override via flag [#6331](https://github.com/GoogleContainerTools/skaffold/pull/6331)
+* Add ContainerTracker for tracking deployed docker containers [#6266](https://github.com/GoogleContainerTools/skaffold/pull/6266)
+* [v2] Enable `skaffold dev` [#6261](https://github.com/GoogleContainerTools/skaffold/pull/6261)
```
Comment (tejal29) [inline CHANGELOG.md:10]: Please leave out v2 related changes from Release notes since users can't use them yet.

Comment (aaron-prindle) [reply CHANGELOG.md:10]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (aaron-prindle) [reply CHANGELOG.md:7]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (aaron-prindle) [reply CHANGELOG.md]: Done

**Comment by aaron-prindle** [comment]
Waiting to cut once #6330 is merged.  Also removed docker deployer related notes from CHANGELOG.md as it is not currently accessible to users

**Comment by aaron-prindle** [comment]
Removed `revert` related commits

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,53 @@
...
+* fix: kubeContext override via flag [#6331](https://github.com/GoogleContainerTools/skaffold/pull/6331)
+* Added developer journey tutorial [#6201](https://github.com/GoogleContainerTools/skaffold/pull/6201)
+* Make skaffold reproducible [#6238](https://github.com/GoogleContainerTools/skaffold/pull/6238)
+
+Fixes:
+* Fix tail false but still log issue [#6299](https://github.com/GoogleContainerTools/skaffold/pull/6299)
+* skip manifest validation for default `kubectl deployer` [#6294](https://github.com/GoogleContainerTools/skaffold/pull/6294)
+* Fix switched container/pod names in app log event [#6215](https://github.com/GoogleContainerTools/skaffold/pull/6215)
+
+Updates and Refactors:
```
Comment (briandealwis) [inline CHANGELOG.md:17]: I wonder if it's worth separating _updates_ from _refactors_

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,53 @@
...
+* Give `logrus.Hook` implementation information about task and subtask [#6313](https://github.com/GoogleContainerTools/skaffold/pull/6313)
+* Add `logrus.Logger` return type on `WithEventContext()` [#6309](https://github.com/GoogleContainerTools/skaffold/pull/6309)
+* Add `logrus` hook for sending `SkaffoldLogEvent`s [#6250](https://github.com/GoogleContainerTools/skaffold/pull/6250)
+
+* Prefix port forward links with `http://` [#6295](https://github.com/GoogleContainerTools/skaffold/pull/6295)
+* fix remote branch lookup [#6269](https://github.com/GoogleContainerTools/skaffold/pull/6269)
```
Comment (briandealwis) [inline CHANGELOG.md:23]: This should go in the _fix_ section?

Comment (aaron-prindle) [reply CHANGELOG.md:23]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (aaron-prindle) [reply CHANGELOG.md:19]: Leaving as is for now, somewhat ambiguous to me at least the difference in some cases

**Comment by aaron-prindle** [comment]
#6330 is merged and the release steps have been updated/rerun to include these the changes there

File: CHANGELOG.md (hunk 2 above)
Comment (briandealwis) [reply CHANGELOG.md:8]: I don't think this was done

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,54 @@
+# v1.29.0 Release - 07/30/2021
+Note: This release comes with a new config version, `v2beta20`. To upgrade your skaffold.yaml, use `skaffold fix`. If you choose not to upgrade, skaffold will auto-upgrade as best as it can.
+
+Highlights:
+* add documentation for `lifecycle hooks` [#6330](https://github.com/GoogleContainerTools/skaffold/pull/6330)
```
Comment (briandealwis) [inline CHANGELOG.md:5]: How about: `Introducing Skaffold lifecycle hooks`?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,54 @@
...
+* Skaffold master -> main [#6263](https://github.com/GoogleContainerTools/skaffold/pull/6263)
+* Add state to event handler to track `logrus` output events [#6272](https://github.com/GoogleContainerTools/skaffold/pull/6272)
+* Remove skaffoldLogs endpoint, consolidate into events endpoint [#6296](https://github.com/GoogleContainerTools/skaffold/pull/6296)
+* fix: kubeContext override via flag [#6331](https://github.com/GoogleContainerTools/skaffold/pull/6331)
+* Added developer journey tutorial [#6201](https://github.com/GoogleContainerTools/skaffold/pull/6201)
+* Make skaffold reproducible [#6238](https://github.com/GoogleContainerTools/skaffold/pull/6238)
```
Comment (briandealwis) [inline CHANGELOG.md:11]: This seems more a fix/update

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,54 @@
+# v1.29.0 Release - 07/30/2021
+Note: This release comes with a new config version, `v2beta20`. To upgrade your skaffold.yaml, use `skaffold fix`. If you choose not to upgrade, skaffold will auto-upgrade as best as it can.
+
+Highlights:
+* add documentation for `lifecycle hooks` [#6330](https://github.com/GoogleContainerTools/skaffold/pull/6330)
+* Skaffold master -> main [#6263](https://github.com/GoogleContainerTools/skaffold/pull/6263)
```
Comment (briandealwis) [inline CHANGELOG.md:6]:
```suggestion
* Rename Skaffold's `master` branch to `main` [#6263](https://github.com/GoogleContainerTools/skaffold/pull/6263)
```

File: CHANGELOG.md (hunk 6 above)
Comment (aaron-prindle) [reply CHANGELOG.md:5]: Done- - now "Introducing Skaffold lifecycle hooks!"

File: CHANGELOG.md (hunk 7 above)
Comment (aaron-prindle) [reply CHANGELOG.md:11]: Put this into `Fix` section

File: CHANGELOG.md (hunk 8 above)
Comment (aaron-prindle) [reply CHANGELOG.md:6]: Done

File: CHANGELOG.md (hunk 6 above)
Comment (aaron-prindle) [reply CHANGELOG.md]: I don't think this was done

**Review by tejal29 — APPROVED** [review]

**Pull request merged**